File: arrays/hm1.py
def twoSum(nums, target):
    hm = {} # create empty dict to store
    for index, value in enumerate(nums):
        diff = target - nums[index]
        if diff in hm:
            return [hm[diff], index]
        hm[value] = index

print(twoSum([2,7,9], 9))

File: hm1.rb
def hm(nums, target)
  h = Hash.new
  nums.each_with_index do |item, index|
    i = h[target - item]
    return [i, index] if i != nil
    h[item] = index
  end
end

puts hm [5,4,5,3,], 8
  
